src/main/java/net/dv8tion/jda/api/utils/cache/CacheFlag.java

```diff
@@ -105,4 +107,20 @@ public boolean isPresence()
     {
         return requiredIntent == GatewayIntent.GUILD_PRESENCES;
     }
+
+    /**
+     * Collects all cache flags that require the specified intent.
+     *
+     * @param  requiredIntent
+     *         The required intent or null if no intent is required
+     *
+     * @return {@link EnumSet} of the cache flags that require the provided intent
+     */
+    @Nonnull
+    public static EnumSet<CacheFlag> fromRequired(@Nullable GatewayIntent requiredIntent)
+    {
+        EnumSet<CacheFlag> flags = EnumSet.allOf(CacheFlag.class);
+        flags.removeIf(flag -> flag.requiredIntent == requiredIntent);
```

Comment:
i think the logic here is wrong? you remove the CacheFlags which require the intent